pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -337,7 +337,7 @@ func (r *Reconciler) addonReconcileWrapper(
 	result, err := reconcile(ctx)
 
 	// Only set to true if we had no error and don't want to requeue the cluster
-	if err == nil && (result == nil || (!result.Requeue && result.RequeueAfter == 0)) {
```

Comment:
Note for reviewer:

`result.Requeue` has been deprecated and it's not recommended to use it anymore https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Result